src/utils/getEtherscanLink.ts

```diff
@@ -0,0 +1,22 @@
+export function getEtherscanLink(
```

Comment:
this can be greatly simplified by changing the `transaction` type to `tx`:
```js
export function getEtherscanLink(
  data: string,
  type: "tx" | "token" | "address" | "block",
): string {
  return `https://etherscan.io/${type}/${data}`
}
```